### PR TITLE
Add default value to use-media-query for SSR

### LINKED ISSUE #1230: Hydration mismatch after SSR using useMediaQuery
### What package has an issue

@mantine/hooks

### Describe the bug

Hello, I have had an incredible time using Mantine and really appreciate all the incredible work the team has done putting this together for the community!

I came across this issue using useMedaQuery from @mantine/hooks that I was familiar with from another package. I wanted to point this out here in case support for SSR is on the roadmap. Here is repro using nextjs

```sh
npx create-next-app@latest
```

index.tsx
```typescript
import type { NextPage } from 'next'
import { useMediaQuery } from '@mantine/hooks'

const Home: NextPage = () => {
  const md = useMediaQuery('(min-width: 1000px)')
  console.log(md, typeof window === 'undefined' ? 'server' : 'client')
  return <span>{String(md)}</span>
}

export default Home
```

Screen Shot 2022-04-11 at 17 32 59

The SSR of useMediaQuery has an initial state of false where the client defaults to the result of the media query: true

### In which browser did the problem occur

chrome

### If possible, please include a link to a codesandbox with the reproduced problem

_No response_

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

_No response_

### Possible fix

Here is some more context around a fix for this. https://github.com/streamich/react-use/pull/2216

### PR DESCRIPTION
**Pull request by teaguestockwell**
Closes #1230 using the [same fix for react-use/useMedia](https://github.com/streamich/react-use/pull/2216)

### CODE REVIEW

File: docs/src/docs/hooks/use-media-query.mdx [hunk 1]
```diff
@@ -19,7 +19,9 @@ use-media-query hook allows to subscribe to media queries.
 It receives media query as an argument and returns true
 if given media query matches current state.
 Hook relies on `window.matchMedia()` [API](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia)
-and will always return false if api is not available (e.g. during server side rendering).
+and will return false if api is not available.
+When server side rendering, it is important to pass ssrInitialValue because without it the server's initial state will fallback to false, but the client will initialize to the result of the media query. 
+When React hydrates the server render, it may not match the client's state. See the [React docs](https://reactjs.org/docs/react-dom.html#hydrate) for more on why this is can lead to costly bugs 🐛.
```
Comment (rtivital) [inline docs/src/docs/hooks/use-media-query.mdx:24]: Do not use emoji.

File: src/mantine-hooks/src/use-media-query/use-media-query.ts [hunk 2]
```diff
@@ -16,15 +16,27 @@ function attachMediaListener(query: MediaQueryList, callback: MediaQueryCallback
   }
 }
 
-function getInitialValue(query: string) {
+function getInitialValue(query: string, ssrInitialValue?: boolean) {
+  // server side render must have a default value to prevent a React hydration mismatch
```
Comment (rtivital) [inline src/mantine-hooks/src/use-media-query/use-media-query.ts:20]: Remove meaningless comments (all of them).

File: docs/src/docs/hooks/use-media-query.mdx [hunk 3]
```diff
@@ -19,7 +19,9 @@ use-media-query hook allows to subscribe to media queries.
 It receives media query as an argument and returns true
 if given media query matches current state.
 Hook relies on `window.matchMedia()` [API](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia)
-and will always return false if api is not available (e.g. during server side rendering).
+and will return false if api is not available.
+When server side rendering, it is important to pass ssrInitialValue because without it the server's initial state will fallback to false, but the client will initialize to the result of the media query.
```
Comment (rtivital) [inline docs/src/docs/hooks/use-media-query.mdx:23]: It says to pass ssrInitialValue value, but does not show how to do that with code. Server side rendering must be described in a separate section.

**Review by rtivital — CHANGES_REQUESTED** [review]

File: docs/src/docs/hooks/use-media-query.mdx (hunk 3 above)
Comment (teaguestockwell) [reply docs/src/docs/hooks/use-media-query.mdx]: Added ssr section to docs

File: docs/src/docs/hooks/use-media-query.mdx [hunk 4]
```diff
@@ -19,9 +19,18 @@ use-media-query hook allows to subscribe to media queries.
...
 <Demo data={HooksDemos.useMediaQueryDemo} />
+
+## Server Side Rendering
+When rendering a component on the server that uses useMediaQuery, it is important to pass the second argument `ssrInitialValue` because without it the server may generate html that the React client cannot correctly hydrate. 
+See the [React docs](https://reactjs.org/docs/react-dom.html#hydrate) for more on why this is can lead to costly bugs.
+
+For example, the server will generate the html for clients using a smaller device.
+After the React client hydrates this, the hook will render again and return the result of the media query:
+
+<Demo data={HooksDemos.useMediaQuerySSRDemo} />
```
Comment (rtivital) [inline docs/src/docs/hooks/use-media-query.mdx:36]: There is no need to include it as a demo, showing just code will be enough

File: src/mantine-hooks/src/use-media-query/use-media-query.ts [hunk 5]
```diff
@@ -16,15 +16,25 @@ function attachMediaListener(query: MediaQueryList, callback: MediaQueryCallback
...
+  console.error(
+    '[@mantine/hooks] use-media-query: Please provide a default value when using server side rendering to prevent a hydration mismatch.'
+  );
+
   return false;
 }
 
-export function useMediaQuery(query: string) {
-  const [matches, setMatches] = useState(getInitialValue(query));
+export function useMediaQuery(query: string, ssrInitialValue?: boolean) {
```
Comment (rtivital) [inline src/mantine-hooks/src/use-media-query/use-media-query.ts:36]: Let's rename `ssrInitialValue` to `initialValue`

**Review by rtivital — CHANGES_REQUESTED** [review]

File: src/mantine-hooks/src/use-media-query/use-media-query.ts (hunk 5 above)
Comment (teaguestockwell) [reply src/mantine-hooks/src/use-media-query/use-media-query.ts]: Fair enough, I renamed these.

**Comment by rtivital** [comment]
@teaguestockwell please fix errors

**Comment by rtivital** [comment]
Thanks!

**Comment by teaguestockwell** [comment]
@rtivital, It's the very least I can do, your work on Mantine has been such a force multiplier! Thank you!

**Pull request merged**